climada/engine/uncertainty/base.py

```diff
@@ -0,0 +1,771 @@
+                        "in sample. Note that the sensitivity analysis will " 
+                        "then return Nan. " 
+                        "See https://github.com/SALib/SALib/issues/237")
+        return check
+        
+    
+    @property
+    def n_samples(self):
+        """
+        The effective number of samples
+
+        Returns
+        -------
+        n_samples: int
+            effective number of samples
+
+        """
+
+        if isinstance(self.samples_df, pd.DataFrame):
+            return self.samples_df.shape[0]
+        return 0
+
+    @property
+    def param_labels(self):
+        """
+        Labels of all uncertainty parameters.
+
+        Returns
+        -------
+        param_labels: list of strings
+            Labels of all uncertainty parameters.
+
+        """
+        return list(self.distr_dict.keys())
+
+    @property
+    def distr_dict(self):
+        """
+        Dictionary of the distribution of all the parameters of all variables
+        listed in self.unc_vars
+
+        Returns
+        -------
+        distr_dict : dict( sp.stats objects )
+            Dictionary of all distributions.
+
+        """
+
+        distr_dict = dict()
+        for unc_var in self.unc_vars.values():
+            distr_dict.update(unc_var.distr_dict)
+        return distr_dict
+    
+    @property
+    def problem(self):
+        """
+        The description of the uncertainty variables and their
+        distribution as used in SALib.
+        https://salib.readthedocs.io/en/latest/getting-started.html.
+
+        Returns
+        -------
+        problem : dict
+            Salib problem dictionnary.
+
+        """
+        return {
+            'num_vars' : len(self.param_labels),
+            'names' : self.param_labels,
+            'bounds' : [[0, 1]]*len(self.param_labels)
+            }
+
+
+    def make_sample(self, N, sampling_method='saltelli',
+                          sampling_kwargs=None):
+        """
+        Make a sample for all parameters with their respective
+        distributions using the chosen sampling_method from SALib.
+        https://salib.readthedocs.io/en/latest/api.html
+
+        Parameters
+        ----------
+        N : int
+            Number of samples as defined in SALib.sample.saltelli.sample().
+        sampling_method: string
+            The sampling method as defined in SALib. Possible choices:
+            'saltelli', 'fast_sampler', 'latin', 'morris', 'dgsm', 'ff'
+            https://salib.readthedocs.io/en/latest/api.html
+            The default is 'saltelli'
+        sampling_kwargs: dict()
+            Optional keyword arguments of the chosen SALib sampling method.
+
+
+        Returns
+        -------
+        df_samples: pd.DataFrame()
+            Dataframe of the generated samples 
+            (one row = one sample, columns = uncertainty parameters)
+        """
+        
+        self.samples_df = sampling_method
```

Comment:
do you need the assignment at line 358? why is the df getting a string? Also I do not think is used before being reassigned at line 367

Reply:
Good catch, this line does not belong there.